tests/sync/partial_sync.cpp

```diff
@@ -231,11 +246,12 @@ TEST_CASE("Partial sync", "[sync]") {
         });
     }
```

Comment:
It looks like we're missing  tests for new functionality that's been added: query names (both user-provided and our automatically created ones), removal of subscriptions (is this implemented in this PR?), etc.

Reply:
Removal of a subscription was not part of this PR since removing it is the equivalent of just an asynchronous write transaction, and since there are differences in how the bindings do async, it wasn't clear that adding it to Object Store would be helpful? 

Perhaps adding a synchronous method for doing it, and then letting it be up to bindings to make sure to call this method in whatever way the run asynchronous write transactions? At least the logic for finding and deleting the subscription will be in one place

Reply:
The logic for finding the object to delete, error handling, etc. should really be in object store so bindings don't have to reinvent it.